src/url.rs

```diff
@@ -314,6 +353,45 @@ impl PyMultiHostUrl {
     fn __getnewargs__(&self) -> (String,) {
         (self.__str__(),)
     }
+
+    #[classmethod]
+    #[pyo3(signature=(*, scheme, host, user=None, password=None, port=None, path=None, query=None, fragment=None))]
+    #[allow(clippy::too_many_arguments)]
+    pub fn build(
+        _cls: &PyType,
+        scheme: &str,
+        host: &str,
+        user: Option<&str>,
+        password: Option<&str>,
+        port: Option<&str>,
```

Comment:
I think we want to accept multiple host/user/password/port combinations in `MutiHostUrl.build`. Potentially easiest to accept these as `MultiHostHost` arguments (see `__init__.py`).

On the Rust side that would potentially be `hosts: Vec<&PyDict>` where you could then extract each host in turn.

Reply:
Interestingly again the existing behaviour was that Url/MultiHostUrl had the same `.build()` behaviour in V1, so thats what I've done here. Would be happy to change if that's what we think needs to be done! 🙂